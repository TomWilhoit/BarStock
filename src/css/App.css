/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

