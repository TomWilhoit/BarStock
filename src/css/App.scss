/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
