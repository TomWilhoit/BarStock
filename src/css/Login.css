
.Login-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  height: 100%;
  width: 100%;
  position: absolute;
  background-image: url(./beer.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: top center;
  z-index: 9999999999;
}

.Login {
  margin-right: auto;
  padding: 0 5%;
}


.login-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.login-quote {
  font-family: 'Abel', sans-serif;
  color:#fff;
  font-size: 25px;
  text-align: center;
}

.username-input {
  margin-top: 10px;
}

.password-input {
  margin: 7px 0px 7px 0px;
}

.bar-logo {
  display: flex;
  justify-content: left;
  width: 100%;
  max-width: 750px;
}