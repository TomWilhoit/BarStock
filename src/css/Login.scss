
.Login-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  height: 100%;
  width: 100%;
  position: absolute;
  background-image: url(./beer.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: top center;
  z-index: 9999999999;
}

.Login {
  margin-right: auto;
  padding: 0 5%;
}

.bar-logo {
  max-width: 350px;
}

.login-quote {
  margin: 1em 0;
  font-family: 'Abel', sans-serif;
  color:#fff;
  font-size: 25px;
  /* text-align: center; */
}

.login-form {
  display: flex;
  flex-direction: column;
  align-items: space-between;
  justify-content: center;
}

.username-input,
.password-input,
.login-button {
  font-size: 1rem;
  border: none;
}

.username-input,
.password-input {
  margin-bottom: 10px;
  padding: 5px;
  height: 25px;
  color: #757575;
}

.username-input,
.password-input:focus {
  outline: none;
}

.login-button {
  height: 40px;
  font-size: 1rem;
  background-color:	#EEAD0E;
  color: #fff;
}

.login-button:hover {
  background-color:	#F8DE9E;
}
