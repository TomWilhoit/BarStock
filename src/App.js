import React, { Component } from 'react';
import MockData from './mockData.js'
import './css/App.css';
import Inventory from './Inventory.js';
import Totals from './Totals.js';
import Header from './Header.js';
import Footer from './Footer.js';
import Login from './Login.js';



class App extends Component {
  constructor() {
    super();
    this.state = {
      allProducts: [MockData.distributor[0].inventory], 
      allMenu: [MockData.menu],
      cartItems: [],
      totalCost: 0,
      totalProjected: 0,
      currentUser: ""
    }
  }
  toggleAlert() {
    alert('toggle');
  }
  render() {
    return (
      <div className="App">
        {/* <div className="Login-container">
          <Login/>
        </div> */}
        <div className="Header-container">
          <Header/>
        </div>
        <div className="Inventory-container">
          <Inventory allProducts={this.state.allProducts}/>
        </div>
        <div className="Totals-container">
          <Totals allProducts={this.state.allProducts} 
                  allMenu={this.state.allMenu} />
        </div>
        <div className="Footer-container">
          <Footer/>
        </div>
      </div>
    )
  }
}

export default App;
