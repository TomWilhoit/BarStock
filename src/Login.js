import React, { Component } from 'react';
import MockData from './mockData.js';
import App from './App.js';
import BarLogo from './BarStock_Logo.png'
import './css/Login.scss';


class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userNameValue : "" ,
      passwordValue : ""
    }
  }

  handleEvent = (event) => {
    const { name, value } = event.target
    this.setState({
      [name] : value
    })
    this.props.loginAccount(this.state.userNameValue)
  }

  handleClick = (e) => { 
    e.preventDefault();
    this.props.toggleLogin();

  }

  render() {
    return (
      <div className="Login-container"> 
        <form className="Login">
          <img className="bar-logo" src={BarLogo} alt="Bar Stock"/>
          <p className="login-quote">Making inventory easy for bars, both big and small.</p>
          <div className="login-form">
            <input id="userNameID" className="username-input" type="text" value={this.state.userNameValue} placeholder="Username" onChange={this.handleEvent} name="userNameValue"></input>
            <input className="password-input" type="password" value={this.state.passwordValue} name="passwordValue" placeholder="Password" onChange={this.handleEvent}></input>
            <button className="login-button" onClick={this.handleClick}>Let's get started!</button>
          </div>
        </form>
      </div> 
    )
  } 
}






export default Login;