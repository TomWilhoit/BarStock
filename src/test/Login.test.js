import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import Login from '../Login';

const mockFunc = jest.fn();

describe('Login', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Login 
        handleEvent = {mockFunc}
        handleClick = {mockFunc}  
        toggleLogin = {mockFunc}
        loginAccount = {mockFunc}
      />
      )
    });

    it('should match snapshot when all data is passed correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should update the state of userNameValue and passwordValue on change', () => {
      const initialState = {userNameValue: '', passwordValue: ''}
      const expectedState = {userNameValue: 'Taylor', passwordValue: 'abc'}
      expect(wrapper.state()).toEqual(initialState);
      wrapper.find('.username-input').simulate('change', {'target':{name:'userNameValue',value:'Taylor'}});console.warn();
      
      wrapper.find('.password-input').simulate('change', {'target':{name:'passwordValue',value:'abc'}});
      expect(wrapper.state()).toEqual(expectedState)
    });

    it('should invoke toggleLogin', () => {
      wrapper.find('.login-button').simulate('click', {preventDefault: () => {}})
      expect(mockFunc).toBeCalled();
    })
    
  })