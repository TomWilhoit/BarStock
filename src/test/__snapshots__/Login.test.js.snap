// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot when all data is passed correctly 1`] = `
<div
  className="Login-container"
>
  <form
    className="Login"
  >
    <img
      alt="Bar Stock"
      className="bar-logo"
      src="BarStock_Logo.png"
    />
    <p
      className="login-quote"
    >
      Making inventory easy for bars, both big and small.
    </p>
    <div
      className="login-form"
    >
      <input
        className="username-input"
        id="userNameID"
        name="userNameValue"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        className="password-input"
        name="passwordValue"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="login-button"
        onClick={[Function]}
      >
        Let's get started!
      </button>
    </div>
  </form>
</div>
`;
